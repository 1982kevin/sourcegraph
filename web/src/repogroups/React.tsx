import { RepogroupMetadata } from './types'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import * as React from 'react'
export const react: RepogroupMetadata = {
    title: 'React',
    name: 'react',
    url: '/react',
    description:
        "Explore React repositories on GitHub. The search examples show how to use React Hook useState with various data types.",
    examples: [
        {
            title: 'Find imports of useState with regex search',
            exampleQuery: <>import [^;]+useState[^;]+ from 'react'</>,
            rawQuery: "import [^;]+useState[^;]+ from 'react'",
            patternType: SearchPatternType.regexp,
        },
        {
            title: 'Examples of useState with an object as the input parameter',
            description: `useState takes a single argument and can accept a primitive, an array or an object.
            Syntax for usage is [state_value, function_to_update_state_value] = useState (initial_state_value).`,
            exampleQuery: <>{'useState({:[string]})'}</>,
            rawQuery: 'useState({:[string]}) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter',
            exampleQuery: <>useState(:[string])</>,
            rawQuery: 'useState(:[string]) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter in only TypeScript files',
            exampleQuery: (
                <>
                    useState(:[string])
                    <span className="search-keyword">lang:</span>typescript
                </>
            ),
            rawQuery: 'useState(:[string]) count:1000 lang:typescript',
            patternType: SearchPatternType.structural,
        },
    ],
    homepageDescription: 'Learn React with code examples.',
    homepageIcon: 'https://cdn4.iconfinder.com/data/icons/logos-3/600/React.js_logo-512.png',
}
