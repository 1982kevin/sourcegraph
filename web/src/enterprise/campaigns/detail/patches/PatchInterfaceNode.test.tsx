import * as H from 'history'
import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { PatchInterfaceNode } from './PatchInterfaceNode'
import { Subject } from 'rxjs'
import { PatchInterface } from '../../../../../../shared/src/graphql/schema'

describe('PatchInterfaceNode', () => {
    const history = H.createMemoryHistory({ keyLength: 0 })
    const location = H.createLocation('/campaigns')
    test('renders HiddenPatch', () => {
        const renderer = createRenderer()
        renderer.render(
            <PatchInterfaceNode
                isLightTheme={true}
                history={history}
                location={location}
                campaignUpdates={new Subject<void>()}
                enablePublishing={false}
                node={{ __typename: 'HiddenPatch' } as PatchInterface}
            />
        )
        const result = renderer.getRenderOutput()
        expect(result.props).toMatchSnapshot()
    })
    test('renders Patch', () => {
        const renderer = createRenderer()
        renderer.render(
            <PatchInterfaceNode
                isLightTheme={true}
                history={history}
                location={location}
                campaignUpdates={new Subject<void>()}
                enablePublishing={false}
                node={{ __typename: 'Patch' } as PatchInterface}
            />
        )
        const result = renderer.getRenderOutput()
        expect(result.props).toMatchSnapshot()
    })
})
