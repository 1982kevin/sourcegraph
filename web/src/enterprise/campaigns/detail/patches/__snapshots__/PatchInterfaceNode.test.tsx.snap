// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatchInterfaceNode renders HiddenPatch 1`] = `Object {}`;

exports[`PatchInterfaceNode renders Patch 1`] = `
Object {
  "campaignUpdates": Subject {
    "_isScalar": false,
    "closed": false,
    "hasError": false,
    "isStopped": false,
    "observers": Array [],
    "thrownError": null,
  },
  "enablePublishing": false,
  "history": Object {
    "action": "POP",
    "block": [Function],
    "canGo": [Function],
    "createHref": [Function],
    "entries": Array [
      Object {
        "hash": "",
        "key": undefined,
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
    ],
    "go": [Function],
    "goBack": [Function],
    "goForward": [Function],
    "index": 0,
    "length": 1,
    "listen": [Function],
    "location": Object {
      "hash": "",
      "key": undefined,
      "pathname": "/",
      "search": "",
      "state": undefined,
    },
    "push": [Function],
    "replace": [Function],
  },
  "isLightTheme": true,
  "location": Object {
    "hash": "",
    "key": undefined,
    "pathname": "/campaigns",
    "search": "",
    "state": undefined,
  },
  "node": Object {
    "__typename": "Patch",
  },
}
`;
